geoh5py/ui_json/enforcers.py

```diff
@@ -0,0 +1,350 @@
+        if enforcer_type == "value":
+            enforcer = ValueEnforcer(validation)  # type: ignore
+        if enforcer_type == "uuid":
+            enforcer = UUIDEnforcer(validation)  # type: ignore
+        if enforcer_type == "required":
+            enforcer = RequiredEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_uijson_parameters":
+            enforcer = RequiredUIJsonParameterEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_form_members":
+            enforcer = RequiredFormMemberEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_workspace_objects":
+            enforcer = RequiredWorkspaceObjectEnforcer(validation)  # type: ignore
+        if enforcer_type == "required_object_data":
+            enforcer = RequiredObjectDataEnforcer(validation)  # type: ignore
+
+        return enforcer
+
+    def validate(self, value: Any):
+        """Validate value against all enforcers."""
+
+        for enforcer in self.enforcers:
+            self._capture_error(enforcer, value)
+
+        self._raise_errors()
+
+    def _capture_error(self, enforcer: Enforcer, value: Any):
+        """Catch and store 'BaseValidationError's for aggregation."""
+        try:
+            enforcer.enforce(self.name, value)
+        except BaseValidationError as err:
+            self._errors.append(err)
+
+    def _raise_errors(self):
+        """Raise errors if any exist, aggregate if more than one."""
+        if self._errors:
+            if len(self._errors) > 1:
+                raise AggregateValidationError(self.name, self._errors)
+            raise self._errors.pop()
+
+
+class Enforcer(ABC):
+    """
+    Base class for rule enforcers.
+
+    :param enforcer_type: Type of enforcer.
+    :param validations: Value(s) to validate parameter value against.
+    """
+
+    enforcer_type: str = ""
+
+    def __init__(self, validations: Any | None = None):
+        self._validations = validations
+
+    @abstractmethod
+    def rule(self, value: Any):
+        """True if 'value' adheres to enforcers rule."""
+        raise NotImplementedError(
+            "Enforcer is abstract. Must be implemented in subclass"
+        )
+
+    @abstractmethod
+    def enforce(self, name: str, value: Any):
+        """Enforces rule on 'name' parameter's 'value'."""
+        raise NotImplementedError(
+            "Enforcer is abstract. Must be implemented in subclass"
+        )
+
+    @property
+    def validations(self):
+        return self._validations
+
+    def __eq__(self, other) -> bool:
+        """Equal if same type and validations."""
+
+        is_equal = False
+        if isinstance(other, type(self)):
+            is_equal = other.validations == self.validations
+
+        return is_equal
+
+    def __str__(self):
+        return f"<{type(self).__name__}> : {self.validations}"
+
+
+class TypeEnforcer(Enforcer):
+    """
+    Enforces type validation.

```

Comment:
```suggestion
    Enforces parameter value type.

```

Reply:
Except it doesn't have to be a parameter.  How about 'Enforcer valid type(s)'